iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c

```diff
@@ -1357,6 +1357,7 @@ static void setup_IoTHubTransport_MQTT_Common_DoWork_events_mocks(
     if (!resend)
     {
         EXPECTED_CALL(gballoc_malloc(IGNORED_NUM_ARG));
```

Comment:
Tests must also be modified to reflect the change in this logic:
https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c#L4916